api/v1/serviceinstance_types.go

```diff
@@ -184,4 +199,21 @@ func init() {
 	SchemeBuilder.Register(&ServiceInstance{}, &ServiceInstanceList{})
 }
 
-func (in *ServiceInstance) Hub() {}
+func (si *ServiceInstance) Hub() {}
+
+func (si *ServiceInstance) getCurrentSharedState() metav1.ConditionStatus {
+	return si.Status.Shared
+}
+
+func (si *ServiceInstance) SharedStateChanged(newShareState *bool, oldShareState metav1.ConditionStatus) bool {
```

Comment:
1. the serviceinstance_types should not have such functions. it should have only setters and getters
2. i think the name is misleading because you are comparing the desired state (spec) with the current state (status)

Reply:
why not? (changed to isSharedDesiredStateChanged)
I use it from the webhook and from the controller - so this is the right place as i see it